pkg/agent/clientset.go

```diff
@@ -210,3 +236,72 @@ func (cs *ClientSet) shutdown() {
 		delete(cs.clients, serverID)
 	}
 }
+
+func (cs *ClientSet) syncPodStatus() {
+	if !cs.enablePodCondition || cs.k8sClient == nil || cs.podNamespace == "" || cs.podName == "" {
+		return
+	}
+
+	pod, err := cs.k8sClient.CoreV1().Pods(cs.podNamespace).Get(cs.podName, metav1.GetOptions{})
+	if err != nil {
+		klog.Warningf("error while getting pod: %s", err)
+		return
+	}
+
+	updatedPod := cs.appendPodCondition(pod, getPodCondition(cs.ClientsCount(), cs.serverCount))
+	if updatedPod == nil {
+		return
+	}
+
+	_, err = cs.k8sClient.CoreV1().Pods(updatedPod.Namespace).UpdateStatus(updatedPod)
```

Comment:
Should we patch instead? I've seen controllers that go both ways so not sure.

Reply:
I would avoid patch (At least the old JSON one). If you can get this working with Server Side Apply, that would probably be an improvement.

Reply:
/cc @jpbetz

Reply:
Let's leave as-is then. I'd like to avoid taking a dependency on SSA for now since some users might still have its release gate disabled.